404 doc link when Gradle fails with 'Resolution of the configuration attempted without an exclusive lock'

### Current Behavior

When a build fails due to 'Resolution of the configuration attempted without an exclusive lock', Gradle logs a link. This link 404s.

```
> For more information, please refer to https://docs.gradle.org/9.0.0-rc-3/userguide/viewing_debugging_dependencies.html.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.
```

Note the repeated suffix: `.html.html`

### Expected Behavior

The link is valid.

### Context (optional)

```
13:22:22   FAILURE: Build failed with an exception.
13:22:22   
13:22:22   * What went wrong:
13:22:22   Execution failed for task ':wasmRootPackageJson'.
13:22:22   > Failed to calculate the value of task ':wasmRootPackageJson' property 'packageJsonFiles'.
13:22:22      > Resolution of the configuration ':sp:wasmJsTestNpmAggregated' was attempted without an exclusive lock. This is unsafe and not allowed.
13:22:22   
13:22:22   * Try:
13:22:22   > For more information, please refer to https://docs.gradle.org/9.0.0-rc-3/userguide/viewing_debugging_dependencies.html.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.
13:22:22   > Run with --debug option to get more log output.
13:22:22   > Get more help at https://help.gradle.org.
13:22:22   
```

### Self-contained Reproducer Project

The invalid links can be found with this search:

https://github.com/search?q=repo%3Agradle%2Fgradle+%2Fviewing_debugging_dependencies.html%5C.html%2F&type=code

### Gradle version

9.0.0-rc-3

### Build scan URL (optional)

_No response_

### Your Environment (optional)

_No response_

--- a/platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r73/DeprecationsCrossVersionSpec.groovy
+++ b/platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r73/DeprecationsCrossVersionSpec.groovy
@@ -58,7 +58,7 @@ class DeprecationsCrossVersionSpec extends ToolingApiSpecification {
         Throwable e = thrown()
         e.message.startsWith("Could not fetch model of type 'List' using connection")
         e.cause.message.contains("Resolution of the configuration ':a:compileClasspath' was attempted without an exclusive lock. This is unsafe and not allowed.")
-        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${targetDist.version.version}/userguide/viewing_debugging_dependencies.html.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
+        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${targetDist.version.version}/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
     }
 
     private void setupBuild() {

--- a/platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/UnsafeConfigurationResolutionDeprecationIntegrationTest.groovy
+++ b/platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/UnsafeConfigurationResolutionDeprecationIntegrationTest.groovy
@@ -61,7 +61,7 @@ class UnsafeConfigurationResolutionDeprecationIntegrationTest extends AbstractDe
         expect:
         fails(":resolve")
         failure.assertHasCause("Resolution of the configuration ':bar:bar' was attempted without an exclusive lock. This is unsafe and not allowed.")
-        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${GradleVersion.current().version}/userguide/viewing_debugging_dependencies.html.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
+        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${GradleVersion.current().version}/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
     }
 
     private String declareRunInAnotherThread() {
@@ -126,7 +126,7 @@ class UnsafeConfigurationResolutionDeprecationIntegrationTest extends AbstractDe
         failure.assertHasFailure("Execution failed for task ':resolve'.") {
             it.assertHasCause("Resolution of the configuration ':bar' was attempted without an exclusive lock. This is unsafe and not allowed.")
         }
-        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${GradleVersion.current().version}/userguide/viewing_debugging_dependencies.html.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
+        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${GradleVersion.current().version}/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
 
         where:
         expression << [
@@ -422,6 +422,6 @@ class UnsafeConfigurationResolutionDeprecationIntegrationTest extends AbstractDe
         executer.withArguments("--parallel", "-I", "init-script.gradle")
         fails(":help")
         failureDescriptionContains("Resolution of the configuration ':foo' was attempted without an exclusive lock. This is unsafe and not allowed.")
-        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${GradleVersion.current().version}/userguide/viewing_debugging_dependencies.html.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
+        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${GradleVersion.current().version}/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
     }
 }

--- a/platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java
+++ b/platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java
@@ -1846,7 +1846,7 @@ public abstract class DefaultConfiguration extends AbstractFileCollection implem
 
         public IllegalResolutionException(String message, DocumentationRegistry documentationRegistry) {
             super(message);
-            resolution = "For more information, please refer to " + documentationRegistry.getDocumentationFor("viewing_debugging_dependencies.html", "sub:resolving-unsafe-configuration-resolution-errors") + " in the Gradle documentation.";
+            resolution = "For more information, please refer to " + documentationRegistry.getDocumentationFor("viewing_debugging_dependencies", "sub:resolving-unsafe-configuration-resolution-errors") + " in the Gradle documentation.";
         }
 
         @Override

--- a/testing/integ-test/src/integTest/groovy/org/gradle/integtests/ParallelStaleOutputIntegrationTest.groovy
+++ b/testing/integ-test/src/integTest/groovy/org/gradle/integtests/ParallelStaleOutputIntegrationTest.groovy
@@ -90,6 +90,6 @@ class ParallelStaleOutputIntegrationTest extends AbstractIntegrationSpec {
         failure.assertThatDescription(Matchers.matchesRegexp("Could not create task ':(a|b):bar'\\."))
         failure.assertHasCause("Could not create task of type 'BadTask'.")
         failure.assertThatCause(Matchers.matchesRegexp("Resolution of the configuration ':(a|b):myconf' was attempted without an exclusive lock\\. This is unsafe and not allowed\\."))
-        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${GradleVersion.current().version}/userguide/viewing_debugging_dependencies.html.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
+        failure.assertHasResolution("For more information, please refer to https://docs.gradle.org/${GradleVersion.current().version}/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors in the Gradle documentation.")
     }
 }

